src/csharp/Microsoft.Spark.UnitTest/Sql/TypesTests.cs

```diff
@@ -45,6 +48,30 @@ public void TestArrayType()
             Assert.Equal("integer", arrayType.ElementType.TypeName);
             Assert.False(arrayType.ContainsNull);
         }
+
+        [Fact]
+        public void TestArrayTypeFromInternal()
+        {
+            {
+                var arrayType = new ArrayType(new IntegerType());
+                Assert.False(arrayType.NeedConversion());
+
+                var expected = new ArrayList(Enumerable.Range(0, 10).ToArray());
+                var actual = (ArrayList)arrayType.FromInternal(expected);
+                Assert.Equal(expected, actual);
+            }
+            {
+                var dateType = new DateType();
+                var arrayType = new ArrayType(dateType);
+                Assert.True(arrayType.NeedConversion());
+
+                var internalDates = new int[] { 10, 100 };
+                Date[] expected =
+                    internalDates.Select(i => (Date)dateType.FromInternal(i)).ToArray();
+                var actual = (ArrayList)arrayType.FromInternal(new ArrayList(internalDates));
+                Assert.Equal(expected, actual.ToArray());
```

Comment:
ditto?

Reply:
these are different objects.  I think this is fine.

Reply:
oops. I meant line 106.

Reply:
got it.  changed in last commit.